### PR TITLE
Legally Distinct Robotic Officer

### PR DESCRIPTION
**Pull request by IProduceWidgets**
## About the PR

Makes zipties stackable
Adds support for stackable cuffs (kinda) in sharedcuffablesystem 
Adds a Tyranny module for borgs
Adds a slowly recharging taser (for borg)
Adds a slowly charging battery (for borg)
Adds a seclite that has the recharging battery (for borg)
Adds a recipe for Tyranny module

image
image
image
image
image

## Why / Balance

Cuff stacking is a nice to have feature, at least for zipties.

I wanted to make a security borg module that uses the unused assets.

I wanted to add something that can make borgs potentially more combat oriented if the stars align.

Creates a pathway for kidnapping/hostages which could be nice if people want that to happen.

## Technical details

Had to add support for stackable cuffs for the borg to be able to use cuffs and have the lingering effect work.
My implementation is surely not ideal. The stackable system does not track individual entities on the stack, so a new one needs to be spawned to split the stack. Ideally this should be handled by getting the prototype and spawning the correct cuff, but I couldn't figure that out, so it just hardcodes making zipties.

I am relatively new to csharp and object oriented programming in general, so if you have comments (which I'm sure you will) please eli5 as much as possible.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

- add: Legally Distinct Robotic Officer Module (Tyranny Module) as Tier 3 research along with tasers!
- tweak: Zipties are now stackable!

### CODE REVIEW

**Comment by LankLTE** [comment]
This shouldn’t have a truncheon as their laws don’t even allow them to use it on anything but nukies and adding it sort of encourages the opposite. 

I’m not really a fan of this being a module at all- I feel it should be implemented as a separate security chassis if at all rather than your nearest service borg being armed to the teeth for some reason. I’ll let some discussion happen on that front though.

**Comment by IProduceWidgets** [comment]
> This shouldn’t have a truncheon as their laws don’t even allow them to use it on anything but nukies and adding it sort of encourages the opposite.

I didn't want to go with stun baton because it also does damage, and at least this is more clear what will happen. Also, it already has a taser which is super broke for stuns anyway. Also, recently there are ways for borgs to get laws that do not require them to be passive. They can be emagged, but also Ion Storms can set their laws to only consider the HoS crew for instance.

> I’m not really a fan of this being a module at all- I feel it should be implemented as a separate security chassis if at all rather than your nearest service borg being armed to the teeth for some reason. I’ll let some discussion happen on that front though.

I have no issue with that. At the moment it is not chassis specific but changing it once a security chassis is added is a one line change.

**Comment by LankLTE** [comment]
> I didn't want to go with stun baton because it also does damage, and at least this is more clear what will happen.

Yeah, but there shouldn't be a way to provide Borgs with something that only serves to be a weapon because you're really only encouraging them to use it when they shouldn't be (because there's relatively few cases where they actually should be). 

> Recently there are ways for borgs to get laws that do not require them to be passive. They can be emagged, but also Ion Storms can set their laws to only consider the HoS crew for instance.

Why would the crew be account for their laws being either illegally or unintentionally modified. That doesn't really make sense.

**Comment by IProduceWidgets** [comment]
They can also use it to beat non-crew, and according to the rules the (acting) captain can declare threats as non-crew.

**Comment by Aisu9** [comment]
does running off of ziptie give them a free hand?

**Comment by IProduceWidgets** [comment]
> does running off of ziptie give them a free hand?

It does not. That that is why I had to adjust the CuffableSystem.

**Comment by moonheart08** [comment]
no secborgs please

**Comment by keronshb** [comment]
Secborgs are another balancing nightmare + it doesn't help the mindset of "borgs vs antags."

**Comment by deltanedas** [comment]
if validhunting modules are a thing it should be its own chassis not for generic borgs

**Comment by CatMagic1** [comment]
I think it'd be a better idea to just have the taser and not the cuffs so it's more of a "HEY YOU! STOP!" module instead of making a fully kitted out sec borg.

**Comment by notquitehadouken** [comment]
add violence module :trollface:

**Comment by notquitehadouken** [comment]
probably should just remove the truncheon and add a separate, low tier researchable, security module for it in the event of nukies or whatever

**Comment by notquitehadouken** [comment]
whoopy dasy

**Comment by VasilisThePikachu** [comment]
solution: officer beepsky

**Comment by IProduceWidgets** [comment]
I'm open to changing the truncheon out for something else (or making a modified truncheon that does something else). I'm not sure what it would be though. I could add a regular gun, but that seems worse. I could add a handheld radio, but I've never seen anyone but salve use them unless I really went out of my way to hand them out. Other ideas?

**Comment by deltanedas** [comment]
this should be multiple modules not 1 big validhunter module with cuffs truncheon and taser

**Comment by IProduceWidgets** [comment]
> this should be multiple modules not 1 big validhunter module with cuffs truncheon and taser

You think? I dunno it feels kinda UI hell to have borg modules with only 1 or 2 items. Remember they dont have backpacks so youre forcing module cycling.

**Comment by IProduceWidgets** [comment]
The handheld radio idea is growing on me instead of the truncheon. Maybe slightly useless right now, but I could see it being nice at a later date with a handheld radio update to add selectable channels and maybe put them in sec loadouts.

**Comment by mirrorcult** [comment]
i really do not want secborgs in any capacity

**Comment by IProduceWidgets** [comment]
Can we unpack that a bit?

Why embargo them when we can reasonably expect MALF ai, syndicate borg modules, and ai/law upload?
Are borg cops really that different from humanoid cops (aside from the borg ui differences)?

**Comment by mirrorcult** [comment]
i am against them because we had them in SS13 for almost a decade and they universally fucking sucked in every implementation and most servers ended up outright removing them

**Comment by IProduceWidgets** [comment]
What made them suck?
Was it that some law sets don't make sense with them?
Was it that the crew didn't have a counter to them (iirc ss13 borgs were much more tanky)?
Was it something else unaddressed/addressable?

**Comment by mirrorcult** [comment]
they sucked because fundamentally the idea of a borg archetype whose whole deal is to go around validhunting is not fun

**Comment by deltanedas** [comment]
> the idea of a borg archetype whose whole deal is to go around validhunting is not fun

the idea of a role whose whole deal is to go around validhunting is not fun

:trollface:

**Comment by HoidC** [comment]
> > the idea of a borg archetype whose whole deal is to go around validhunting is not fun
> 
> the idea of a role whose whole deal is to go around validhunting is not fun
> 
> :trollface:

their job is to keep order not hunt valids, that's part of the job description but like in truth 30 percent of it

**Comment by Carolyn3114** [comment]
> i am against them because we had them in SS13 for almost a decade and they universally fucking sucked in every implementation and most servers ended up outright removing them

this isn't ss13 and ss13 has bad game design so obviously we need this

**Comment by keronshb** [comment]
> What made them suck? Was it that some law sets don't make sense with them? Was it that the crew didn't have a counter to them (iirc ss13 borgs were much more tanky)? Was it something else unaddressed/addressable?

Mirror got it down pretty well.

Law set didn't matter, just by merely existing it stoked the flames of the "vs antag" mindset instead of "I need to listen to all crew" mindset.

Servers removed them or straight up locked them behind red alert + huge nerfs.

Crews could have the hardest of hard counters and it still was absolutely awful to deal with.

It's not something that can be addressable other than not allowing them at all.

**Comment by IProduceWidgets** [comment]
Okay, Truncheon 1984ed in favor of a working handheld radio (real) and a security borg creed that reinforces that they should do the other 70% of the game.

**Comment by Eden077** [comment]
I think even by the off **chance** these do get added, remove the ability for sci to have access to the module, it should be in the secfab

**Comment by IProduceWidgets** [comment]
I don't like that as a first pass because no other module is in any other fab so its confusing, especially so because there is no sec borg chassis. It can of course be used as a balance change if they prove to be too common because all that would seem to do is make actually producing them more difficult.

I also am not a fan because it makes using the module as an antag much more difficult, but that presumably is not the main point of contention people have with sec borgs.

**Comment by KompaniLinge** [comment]
I'd just like to add my own 2 cents to this PR and comments about it;

To me when I think borgs I think uniform lawset for all borgs, and a overarching idea of a "asimov"-esq system. A security borg goes against that. I don't think its crazy to say less-than-lethal (otherwise known as nonlethal) tactics goes against the whole point of the most principle part of the laws which is "do not harm crew"/"prevent crew harm". 
If we want an actually interesting borg to keep the peace, why not just port the peacekeeper? 

And if we are looking at it from an entirely game balancing POV, this is basically just making sec ghost-roles, which would make the whole point of dying as a security officer a lot more trivial especially when you can get le funny instastun taser with AA and zipties.

**Comment by Kadeo64** [comment]
The thing about this is it's basically never going to be researched. Science, right now, at least on LRP, will never research arsenal tier 3 since 
A: it doesn't benefit the scientists aside from salvage weapons and explosive technology
B: it actively makes antagonist's jobs harder 
(Of course, the exception is nukies.)
So you basically won't ever see this outside of stalling nukie rounds where science has extra materials to convert a roundstart borg into a nukie destroyer. 
So, outside of nukie rounds, and the occassional (see: [byond the impossible](https://tgstation13.org/wiki/Byond_the_impossible)) zombie/rev round where science is able to grind out tier 3 arsenal because HOS said they'll get some cool guns from it, this module will basically never get seen. 
I think it wouldn't affect game balance TOO much outside of these rare situations, but honestly, I would rather have the Peacekeeper borg.

**Comment by LankLTE** [comment]
I don't believe we currently want secborgs in the game. This also would need a unique sprite rather than re-using the service borg, and probably a design document should this be considered in the future.

**Comment by DangerRevolution** [comment]
@IProduceWidgets do the stackable zipties require the C# changes or is that just for the handcuffs to be stackable?
